js/ui/codingManager.js

```diff
@@ -462,6 +472,18 @@ const CodingManager = new Lang.Class({
         }
     },
 
+    _overviewHiding: function(overview, session) {
+        session.buttonApp.show();
+        if (session.buttonBuilder)
+            session.buttonBuilder.show();
+    },
+
+    _overviewShowing: function(overview, session) {
+        session.buttonApp.hide();
+        if (session.buttonBuilder)
+            session.buttonBuilder.hide();
+    },
```

Comment:
I know that there's a few other places where we have to do this as well - is it possible to combine this all into one function? (e.g. `showCodeviewButtons` / `hideCodeviewButtons`) and connect the signals to those or thunks which call those? There's a bit of logic here in terms of checking that `buttonBuilder` is truthy that is getting repeated.